src/components/IconButton/index.tsx

```diff
@@ -1,12 +1,46 @@
 import React, { forwardRef } from 'react';
-import PropTypes from 'prop-types';
 import cc from 'classcat';
 import baseElement from 'util/baseElement';
+import { IconProps } from 'components/Icon';
 
 export const THEMES = ['ghost', 'aqua', 'outlined'];
 export const RADII = ['square', 'circle'];
 export const SIZES = ['s', 'm'];
 
+export interface IconButtonProps {
+  /**
+   * Takes in a react-router `Link` reference and sets it
+   * as the base element. You may ONLY use it like the
+   * following:
+   *
+   * - `import { Link } from 'react-router'`
+   * - `Link={Link}`
+   */
+  Link?: React.ElementType;
```

Comment:
Have you tested this? Also, is there maybe some sort of type definition we could pull in from react-router?

Reply:
We can do something like `Link?: React.ElementType<LinkProps>;`, we'd just need `react-router`'s types, I'll try it out.

Reply:
Hmmm actually now that I think about it 😅  We're really far behind in terms of react router versions. That may not be the best idea. I dunno.